magento/pwa-studio
# [feature]: Store Switcher Groups

**! Only useable after 2.4.2 merge !**

## Description

Differentiate between stores in different groups.
Display "group name - view name" when multiple groups.
Display "view name" when only single group.

## Related Issue





Closes [PWA-1070](https://jira.corp.magento.com/browse/PWA-1070).

## Acceptance



### Verification Stakeholders

### Specification



### Verification Steps


1. Use a 2.4.2 backend.
2. Set up multiple stores (groups)
3. Set up multiple views within each store
4. Visit ui and ensure switcher conforms to ux mocks
5. Ensure switcher has proper overflow when many groups/views are configured.
6. Ensure switcher has proper truncation when groups/views have long names.

## Screenshots / Screen Captures (if appropriate)
[![Image from Gyazo](https://i.gyazo.com/8a4fc6f2ffe60e12c496f3bc5ac1f40b.png)](https://gyazo.com/8a4fc6f2ffe60e12c496f3bc5ac1f40b)

[![Image from Gyazo](https://i.gyazo.com/c0b87e0d6cb8860c5ccded012a42f7d7.png)](https://gyazo.com/c0b87e0d6cb8860c5ccded012a42f7d7)


Truncated long text in mobile: ![](https://i.gyazo.com/8ed1bfeb749823013695e4eb930d2e81.png)

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.
